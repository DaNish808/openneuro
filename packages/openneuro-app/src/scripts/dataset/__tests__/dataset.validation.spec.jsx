import React from 'react'
import { shallow } from 'enzyme'
import Validation from '../dataset.validation.jsx'

describe('dataset/Validation', () => {
  const defProps = {
    errors: [],
    warnings: [],
    validating: false,
    display: true,
  }
  it('renders successfully', () => {
    const wrapper = shallow(<Validation {...defProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders successfully when validating', () => {
    const wrapper = shallow(<Validation {...defProps} validating={true} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('displays nothing with display set to false', () => {
    const wrapper = shallow(<Validation {...defProps} display={false} />)
    expect(wrapper.html()).toBe(null)
  })
})
