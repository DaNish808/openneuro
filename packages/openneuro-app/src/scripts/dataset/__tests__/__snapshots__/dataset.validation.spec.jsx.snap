// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/Validation renders successfully 1`] = `
<div
  className="fade-in col-xs-12 validation"
>
  <h3
    className="metaheader"
  >
    BIDS Validation
  </h3>
  <PanelGroup
    accordion={true}
    activeKey="2"
    bsClass="panel-group"
    className="validation-wrap"
    onSelect={[Function]}
  >
    <Panel
      bsClass="panel"
      bsStyle="default"
      className="status"
      defaultExpanded={false}
      eventKey="1"
      header={
        <div
          className="super-valid"
        >
          <span
            className="dataset-status ds-success"
          >
            <i
              className="fa fa-check-circle"
            />
             Valid
          </span>
        </div>
      }
    >
      <br />
      <ValidationResults
        errors={Array []}
        warnings={Array []}
      />
    </Panel>
  </PanelGroup>
</div>
`;

exports[`dataset/Validation renders successfully when validating 1`] = `
<div
  className="fade-in col-xs-12 validation"
>
  <h3
    className="metaheader"
  >
    BIDS Validation
  </h3>
  <Spinner
    active={true}
    text="Validating"
  />
</div>
`;
